dolphin-dnn/src/main/java/edu/snu/dolphin/dnn/layers/ConvolutionalLayer.java

```diff
@@ -77,13 +82,87 @@ public boolean isLearnable() {
   }
 
   /**
+   * Transform the given image to column form to facilitate matrix multiplication.
+   * @param imageIndex the index of the image in the input matrix.
+   * @param input input values for this layer.
+   * @return the converted column.
+   */
+  private Matrix im2col(final int imageIndex, final Matrix input) {
+    final int[] inputShape = getInputShape();
+    final Matrix col =
+        matrixFactory.zeros(NeuralNetworkUtils.getShapeLength(outputShape), kernelHeight * kernelWidth);
+    for (int kh = 0; kh < kernelHeight; ++kh) {
```

Comment:
@seyounglee95 and I discussed this issue in an offline meeting, Caffe shows better performance with this implementation. However while Caffe uses row-major order, we use column-major order. This difference may influence performance regarding to memory access pattern.
We will take this issue into account later after considering difference implementations with a micro-benchmark.